nemo_curator/modules/semantic_dedup/clusteringmodel.py

```diff
@@ -29,20 +29,26 @@
 from nemo_curator.log import create_logger
 from nemo_curator.utils.distributed_utils import performance_report_if_with_ts_suffix
 from nemo_curator.utils.file_utils import expand_outdir_and_mkdir
-from nemo_curator.utils.semdedup_utils import assign_and_sort_clusters
+from nemo_curator.utils.semdedup_utils import (
+    L2_DIST_TO_CENT_COL,
+    assign_and_sort_clusters,
+    get_array_from_df,
+    normalize_embeddings_col_in_df,
+)
 
 
-def get_embedding_ar(df: "cudf.DataFrame", embedding_col: str) -> cp.ndarray:
-    return df[embedding_col].list.leaves.values.reshape(len(df), -1)
-
-
-def add_dist_to_cents(
+# TODO : add cosine distance also while we are computing distance to centroids, will allow us for more flexibility in later stages
```

Comment:
Could you open issues for these and link to them here?

Reply:
Fixed in subsequent PR that we merged to https://github.com/NVIDIA/NeMo-Curator/pull/609